tools/purge_omsagent.sh

```diff
@@ -0,0 +1,407 @@
+            echo "Exiting without purging..."
+            call_exit 0
+        else
+            echo "Unknown input. Please type 'y' (yes) or 'n' (no) to proceed."
+        fi
+    done
+
+    # continuing with LAD purge
+    echo "Please go to portal.azure.com and uninstall the LAD extension."
+    echo "Go to Azure Portal -> Virtual Machines -> <vm_name> -> Settings -> Extensions"
+    echo "and then click the '...' in the LinuxDiagnostic row and click 'Uninstall'"
+    read -p "Press enter to proceed once the Extension is uninstalled." toss2
+    echo ""
+
+    # remove LAD directories
+    echo "---------- Removing LAD directories ----------"
+    if [ -d $SOLO_LAD_DIR ]; then
+        rm -rf $SOLO_LAD_DIR
+    fi
+    if [ -d $OMS_LAD_DIR ]; then
+        rm -rf $OMS_LAD_DIR
+    fi
+
+    # remove LAD package
+    echo "---------- Removing LAD package ----------"
+    pkg_rm lad-mdsd
+
+    # remove MDSD package
+    check_if_pkg_is_installed mdsd
+    MDSD_INSTALLED=$?
+    if [ $MDSD_INSTALLED -eq 0 ]; then
+        echo "---------- Removing MDSD package ----------"
+        pkg_rm mdsd
+    fi
+
+    echo "LAD successfully purged!"
+fi
+
+
+# REMOVE OMS
+echo "--------------------------------------------------------------------------------"
+echo "------------------------------ STARTING OMS PURGE ------------------------------"
+echo "--------------------------------------------------------------------------------"
+echo ""
+
+# Check if installed via extension
+ls /var/lib/waagent | grep -i Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux- > /dev/null 2>&1
+EXTENSION_INSTALLED=$?
+if [ $EXTENSION_INSTALLED -eq 0 ]; then
+    echo ""
+    echo "Please go to portal.azure.com and uninstall the OMS extension and its dependencies."
+    echo "Go to Azure Portal -> Virtual Machines -> <vm_name> -> Settings -> Extensions"
+    echo "and then click the '...' in the OMSAgentForLinux row and click 'Uninstall'"
+    echo "(Do this as well with the DependencyAgentLinux row)"
+    read -p "Press enter to proceed once the Extension is uninstalled." toss2
+    echo ""
+fi
+
+# run OMS purge command
+echo "---------- Downloading OMS Onboarding Script ----------"
+if [ ! -f $OMS_ONBOARD_AGENT ]; then
+    wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/master/installer/scripts/onboard_agent.sh
+    if [ $? -ne 0 ]; then
+        echo "Error accessing Github to download the purge script for the OMS Agent" >&2
+        echo "Please ensure that this machine can connect to the following URL:" >&2
+        echo "  https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/master/installer/scripts/onboard_agent.sh" >&2
+        echo "Or ensure the onboard_agent.sh script is available in the same directory as this script." >&2
+        call_exit $ONBOARD_AGENT_MISSING
+    fi
+fi
+echo ""
+
+echo "---------- Running OMS Purge Script ----------"
+chmod +x ./onboard_agent.sh
+./onboard_agent.sh --purge
```

Comment:
this will download the entire bundle again, right? is there some way to not download the entire bundle (will [this](https://github.com/microsoft/OMS-Agent-for-Linux/blob/master/installer/scripts/uninstall.sh) suffice? IDK if this is equal/superset of purge function from `bundle_skel`)

if we have to download the bundle, can you make sure it is deleted afterwards? its 100MB+

Reply:
Actually I think the `uninstall.sh` script you linked would be perfect, I didn't see it before that would work a lot better, thank you!

Reply:
Fixed in this commit: https://github.com/microsoft/OMS-Agent-for-Linux/pull/1255/commits/e7138a75afb34623b64d2811b51f366a35ca4b57